service/storage_proxy.cc

```diff
@@ -3497,8 +3497,40 @@ class abstract_read_executor : public enable_shared_from_this<abstract_read_exec
             });
         });
     }
+    bool maybe_speculatively_fail(digest_read_resolver& resolver, const gms::inet_address& ep, clock_type::time_point timeout) {
+        if (fbu::is_me(ep)) {
+            return false;
+        }
+        if (auto last_seen = _proxy->last_seen(ep)) {
+            auto now = clock_type::now();
+            auto latency_estimate = last_seen->last_sent - last_seen->last_responded;
```

Comment:
Why is this a latency estimate? These are timestamps for unrelated requests. The difference could be negative, zero, or positive.

Reply:
Yeah, the name is unfortunate, it's only a latency estimate in very broad terms with lots of assumption. What I actually want to estimate is "a time period in which the target failed to response", but I don't have any clue how to shorten the name properly.

Reply:
Complicated concepts should have complicated names (or be simplified)